common/v2/features/ReceiveAssets/ReceiveAssets.tsx

```diff
@@ -0,0 +1,255 @@
+
+  const [requestAddress, setRequestAddress] = useState('');
+
+  const initialValues = {
+    amount: '0',
+    asset: { label: 'Ethereum', id: '7bbf42b1-9275-120b-0d0a-a788abd75ea0' },
+    chainId: network ? network.chainId : 1
+  };
+
+  const isValidAmount = (decimal: number) => (amount: string) =>
+    numberIsNotNegative(+amount) && validDecimal(amount, decimal);
+
+  const validateAmount = (amount: any) => {
+    let error;
+    if (selectedAsset) {
+      const { decimal } = selectedAsset;
+      if (decimal && !isValidAmount(decimal)(amount)) {
+        error = 'Please enter a valid amount';
+      }
+    }
+
+    return error;
+  };
+
+  return (
+    <ReceivePanel heading="Receive Assets" icon={receiveIcon} onBack={history.goBack}>
+      <Formik
+        initialValues={initialValues}
+        onSubmit={noop}
+        render={({ values: { amount, chainId }, errors }: FormikProps<typeof initialValues>) => (
+          <Form>
+            <Fieldset>
+              <SLabel htmlFor="recipientAddress">Recipient Address</SLabel>
+              <Field
+                name="recipientAddress"
+                component={({ field, form }: FieldProps) => (
+                  <AccountDropdown
+                    name={field.name}
+                    value={field.value}
+                    accounts={accounts}
+                    onSelect={(option: IExtendedAccount) => {
+                      form.setFieldValue(field.name, option);
+                      setRequestAddress(option.address);
+                    }}
+                  />
+                )}
+              />
+            </Fieldset>
+            <AssetFields>
+              <Amount>
+                <SLabel htmlFor="amount">Amount</SLabel>
+                <Field
+                  name="amount"
+                  validate={validateAmount}
+                  render={({ field, form }: FieldProps<typeof initialValues>) => (
+                    <FullWidthInput
+                      value={field.value}
+                      onChange={({ target: { value } }) => form.setFieldValue(field.name, value)}
+                      placeholder="0.00"
+                    />
+                  )}
+                />
+              </Amount>
+              <Asset>
+                <SLabel htmlFor="asset">Asset</SLabel>
+                <Field
+                  name="asset"
+                  render={({ field, form }: FieldProps<typeof initialValues>) => (
+                    <StyledSelect
+                      name="Assets"
+                      className="select-container"
+                      options={assetOptions}
+                      value={field.value}
+                      onChange={(option: Option) => {
+                        form.setFieldValue(field.name, option);
+                        if (option.label) {
+                          setAssetName(option.label);
+                        }
```

Comment:
_idem_

Reply:
How should I handle using the asset outside of the form for validation (also curious since the yup schema would need this and exists outside of the form as well if I use yup)

https://github.com/MyCryptoHQ/MyCrypto/blob/5cff9078237245171ad89c8dbcaf6fda12d5b538/common/v2/features/ReceiveAssets/ReceiveAssets.tsx#L118